src/pudl_archiver/zenodo/api_client.py

```diff
@@ -0,0 +1,469 @@
+            interface.new_deposition
+        )
+
+        return interface
+
+    async def remote_fileinfo(self, deposition: Deposition):
+        """Return info on all files contained in deposition.
+
+        Args:
+            deposition: Deposition for which to return file info.
+
+        Returns:
+            Dictionary mapping filenames to DepositionFile metadata objects.
+        """
+        url = deposition.links.files
+        params = {"access_token": self.upload_key}
+
+        async with self.session.get(url, params=params) as response:
+            raw_json = await response.json()
+
+        return {file["filename"]: DepositionFile(**file) for file in raw_json}
+
+    async def create_deposition(self, data_source_id: str):
+        """Create a Zenodo deposition resource.
+
+        This should only be called once for a given data source.  The deposition will be
+        prepared in draft form, so that files can be added prior to publication.
+
+        Args:
+            data_source_id: Data source ID that will be used to generate zenodo metadata
+            from data source metadata.
+
+        Returns:
+            Deposition object, per
+            https://developers.zenodo.org/?python#depositions
+        """
+        url = f"{self.api_root}/deposit/depositions"
+        params = {"access_token": self.upload_key}
+        headers = {"Content-Type": "application/json"}
+
+        metadata = DepositionMetadata.from_data_source(data_source_id)
+        data = json.dumps(
+            {
+                "metadata": metadata.dict(
+                    by_alias=True,
+                    exclude={"publication_date", "doi", "prereserve_doi"},
+                )
+            }
+        )
+
+        async with self.session.post(
+            url, params=params, data=data, headers=headers
+        ) as response:
+            # Ignore content type
+            return Deposition(**await response.json(content_type=None))
+
+    async def add_files(self, resources: dict[str, ResourceInfo]):
+        """Add all downloaded files to zenodo deposition.
+
+        This method will check local files against remote files which already
+        exist in Zenodo deposition. If local file name matches remote file name,
+        checksums will be used to determine if file has changed. It will also
+        create and upload a new datapackage.
+
+        Args:
+            resources: Dictionary mapping filenames to ResourceInfo objects,
+            which contain local path to resource and working partitions.
+
+        Returns:
+            Updated Deposition object.
+        """
+        for name, resource in resources.items():
+            if name not in self.deposition_files:
+                self.changed = True
+                logger.info(f"Adding {name} to deposition.")
+
+                with open(resource.local_path, "rb") as f:
+                    await self.upload(f, resource.local_path.name)
+            else:
```

Comment:
I'm a little confused by this method. So at this point in `create_archive` we have a dictionary of resource names to `ResourceInfo` objects and we need to the add them to the Zenodo deposition. 

`add_files()` iterates through the resources and adds them to the deposition. I don't understand the case this `else` statement is handling. Do the archives occasionally find duplicate resources? I probably just don't understand the Zenodo workflow well enough.

Reply:
I guess I'm confused as to what a deposition is.

Reply:
I added a new docstring that hopefully makes this more clear. Basically the deposition is a versioned container for uploading files to zenodo and new versions contain all of the files from the previous version by default. So, this function will check if the file already exists, then see if it's been changed and update it if that's the case

Reply:
Oh that makes more sense thank you.